lib/connect/api/assetRequests/create.js

```diff
@@ -111,13 +111,31 @@ const createRequestForProduct = async (client, inputData, productId, assetExtera
 };
 
 const createAssetRequestFromOrder = async (client, data) => {
+  if (!data.customer_external_uid) {
+    const extUid = await lookupTierByExternalId(client, data.customer_external_id, data.hub_id);
+    if (extUid) {
+      /* eslint-disable no-param-reassign */
+      data.customer_external_uid = extUid;
+    }
+  }
+
   if (!data.t1_external_uid) {
-    const extUid = await lookupTierByExternalId(client, data.t1_external_id);
+    const extUid = await lookupTierByExternalId(client, data.t1_external_id, data.hub_id);
     if (extUid) {
       /* eslint-disable no-param-reassign */
       data.t1_external_uid = extUid;
     }
   }
+
+  if (data.t2_external_id) {
```

Comment:
Better to test this way:

if (data.reseller_tiers == 't2t1') ....

We have the reseller_tiers input field that tell us if a tier 2 is involved in the request.